pkg/skaffold/build/cache/lookup.go

```diff
@@ -132,29 +118,39 @@ func (c *cache) lookupLocal(ctx context.Context, hash, tag string, entry ImageDe
 	return needsBuilding{hash: hash}
 }
 
-func (c *cache) lookupRemote(ctx context.Context, hash, tag string, platforms []specs.Platform, entry ImageDetails) cacheDetails {
-	log.Entry(ctx).Debugf("Looking up %s tag when in entry it's %s", tag, entry.Digest)
-	if remoteDigest, err := docker.RemoteDigest(tag, c.cfg, nil); err == nil {
-		// Image exists remotely with the same tag and digest
+func (c *cache) lookupRemote(ctx context.Context, hash, tag string, platforms []specs.Platform) cacheDetails {
+	var cacheHit bool
+	entry := ImageDetails{}
+
+	if digest, err := docker.RemoteDigest(tag, c.cfg, nil); err == nil {
 		log.Entry(ctx).Debugf("Found %s remote", tag)
-		if remoteDigest == entry.Digest {
-			return found{hash: hash}
-		}
+		entry.Digest = digest
+
+		c.cacheMutex.Lock()
+		c.artifactCache[hash] = entry
+		c.cacheMutex.Unlock()
+		return found{hash: hash}
```

Comment:
Hi @ericzzzzzzz @3droj7, 

I have a concern about the lookupRemote function. It appears that the function may always return found when the same tag is reused, even if the digests are different.

To fix this issue, the lookupRemote function should check the digest of the image in the cache. If the digests are different, the function should return needsRemoteTagging.

Is this correct?

Thanks.

Reply:
I submitted this issue.
https://github.com/GoogleContainerTools/skaffold/issues/9248